### DISCUSSION TITLE
Automate renaming the master branch?

### DISCUSSION THREAD

**Discussion by madebyjeffrey** [community/community#23173] [New to GitHub]
<p>As has been in the news recently, renaming the master branch to main is not a bad thing to want to do.</p>
<p>Although it is easy enough to do it manually, it would be rather laborious to do it for all of my repositories.</p>
<p>Is there any way to automate this process at either the repository level or account level?</p>

**Comment by yamiacat** [comment, marked as answer]
<p>Hi there madebyjeffrey! :wave: Welcome to the Community!</p>
<p>GitHub is working towards making this easier as we speak! I believe we’ll be making an announcement, alongside unveiling documentation and possibly tooling around changing the default branch naming in July, although I’m afraid I don’t have an exact timeline.</p>
<p>Some things, such as the ability to host a GitHub Pages site from a non-master default branch, need a little more work before we’re ready.</p>

**Comment by citylights521** [comment]
<p>Hey <a class="mention" href="/yamiacat">@yamiacat</a> / GitHub team!<br>
My team and I are eagerly awaiting these automation tools as well. From what I’ve read online, GitHubs hope was to have tools that made making the changes ‘seamless’ by late summer/early in the new year.<br>
Any updates around timeline?<br>
And where’s the best place to track progress/make sure we know as soon as the tooling is available?<br>
Thanks!</p>

**Comment by yamiacat** [comment]
<p>Hi <a class="mention" href="/citylights521">@citylights521</a> :wave: Welcome to the Community!</p>
<p>Details of the settings now available and recommended best practice can be found here:</p>
<aside class="onebox githubfolder">
  <header class="source">
      <img src="https://github.githubassets.com/favicons/favicon.svg" class="site-icon" width="32" height="32">
      <a href="https://github.com/github/renaming" target="_blank" rel="noopener">github.com</a>
  </header>
  <article class="onebox-body">
    <div class="aspect-image" style="--aspect-ratio:400/400;"><img src="https://avatars.githubusercontent.com/u/9919?s=400&amp;amp;v=4" class="thumbnail" width="400" height="400"></div>

<h3><a href="https://github.com/github/renaming" target="_blank" rel="noopener">github/renaming</a></h3>


  <p><span class="label1">Guidance for changing the default branch name for GitHub repositories</span></p>

  </article>
  <div class="onebox-metadata">
    
    
  </div>
  <div style="clear: both"></div>
</aside>

<p>They’re aimed at people that want to make the change immediately, and also ensuring that all repos created going forward have the new values.</p>
<p>We’re still working on further tooling that will be introduced to make changing the default branch names of existing repos en mass an easier prospect. When that tooling is ready it will be mentioned in that repository, and it will also be announced on our changelog:</p>
<aside class="onebox allowlistedgeneric">
  <header class="source">
      <img src="https://github.blog/wp-content/uploads/2019/01/cropped-github-favicon-512.png?fit=32%2C32" class="site-icon" width="32" height="32">
      <a href="https://github.blog/changelog/" target="_blank" rel="noopener">The GitHub Blog</a>
  </header>
  <article class="onebox-body">
    <div class="aspect-image" style="--aspect-ratio:690/362;"><img src="https://github.blog/wp-content/uploads/2019/03/blog-card.png?fit=5001%2C2626" class="thumbnail" width="690" height="362"></div>

<h3><a href="https://github.blog/changelog/" target="_blank" rel="noopener">GitHub Changelog</a></h3>



  </article>
  <div class="onebox-metadata">
    
    
  </div>
  <div style="clear: both"></div>
</aside>

**Comment by citylights521** [comment]
<p>This is very insightful <a class="mention" href="/yamiacat">@yamiacat</a>, thank you for the information!<br>
Any sense of rough timing for the further tooling?</p>

**Comment by yamiacat** [comment]
<p>Ah, it turns out it shipped a week or two ago!</p>
<p>Here you go!</p>
<aside class="onebox allowlistedgeneric">
  <header class="source">
      <img src="https://github.blog/wp-content/uploads/2019/01/cropped-github-favicon-512.png?fit=32%2C32" class="site-icon" width="32" height="32">
      <a href="https://github.blog/changelog/2021-01-19-support-for-renaming-an-existing-branch/" target="_blank" rel="noopener">The GitHub Blog</a>
  </header>
  <article class="onebox-body">
    <div class="aspect-image" style="--aspect-ratio:453/381;"><img src="https://user-images.githubusercontent.com/19803303/181075437-bfa296f1-9206-4cac-86dc-a0c6aeff24b5.png" class="thumbnail" width="453" height="381"></div>

<h3><a href="https://github.blog/changelog/2021-01-19-support-for-renaming-an-existing-branch/" target="_blank" rel="noopener">Support for renaming an existing branch - GitHub Changelog</a></h3>

<p>Support for renaming an existing branch</p>


  </article>
  <div class="onebox-metadata">
    
    
  </div>
  <div style="clear: both"></div>
</aside>

**Comment by citylights521** [comment]
<p>AMAZING!!! Thank you!</p>

**Comment by AaronNGray** [comment]
@yamiacat Has GitHub or anyone automated changing all repos in a GitHub user account or Organization from `master` to `main` ?

**Reply by yamiacat** [reply]
@AaronNGray - there was no official tooling introduced to do that.

I did find this script online but haven't tested it, and it's fairly old, so it might need tweaking to update it:

https://andre.arko.net/2020/06/26/automatically-update-every-github-default-branch/

**Reply by AaronNGray** [reply]
Thanks I know bash but really don't know the GitHub API. Is there anyone who could check this out and maybe make it a standard thing ?

**Reply by yamiacat** [reply]
I can add a +1 to our internal feature request tracker but given that it's been a few years since we made the switch from `master` to `main` I don't think it will be prioritized, sorry!

**Reply by AaronNGray** [reply]
I am getting signal from multiple sources from the black community that they are more than very troubled about the usage of the term `master`, I did not really take this as an issue before but now have been pursuaded otherwise due to the nature and time of the horrific abuse concerned. I have a large number of repositories and need to be able to modify them all. I am sure this is the case for a very large number of users, organizations and their repositories and providing an automated feature that can be advertised to GitHub users would go a large way to resolving this issue.
